docs/reference/modules/datatiers.asciidoc

```diff
@@ -0,0 +1,104 @@
+queries and aggregations in a timely manner, as opposed to the <<hot-tier, hot tier>> which
+will prioritise high IO.
+The content tier is a fast, higher cost, tier. The content data has very long data retention
+characteristics and from a resiliency perspective the indices in this tier use one or more
+replicas.
+
+NOTE: new indices that are not part of <<data-streams, data streams>> will be allocated to the
+<<content-tier>>
+
+[[hot-tier]]
+==== Hot tier
+
+The hot tier is made of one or more nodes that have the <<data-hot-node, data_hot>> role.
+It is the {es} entry point for timeseries data. This tier is fast both for reads and writes,
+higher cost as it requires more hardware resources such as SSD drives. Data within this tier
+has retention characteristics that span between days and weeks. From a resiliency perspective
+the indices in this tier use one or more replicas.
+
+NOTE: new indices that are part of a <<data-streams, data stream>> will be allocated to the
+<<hot-tier>>
+
+[[warm-tier]]
+==== Warm tier
+
+The warm tier is made of one or more nodes that have the <<data-warm-node, data_warm>> role.
+This tier is where data goes once it is not queried very frequently in the <<hot-tier, hot tier>>
+anymore. It is a medium-fast, lower cost tier, that still allows data updates. Data within this
+tier is retained for weeks and the indices still have one or more replicas.
+
+The tradeoff when moving data from the Hot to Warm is search and ingest speed for costs.
+
+[[Cold-tier]]
+==== Cold tier
+
+The cold tier is made of one or more nodes that have the <<data-cold-node, data_cold>> role.
+Once the data in the <<warm-tier, warm tier>> is not updated anymore it can transition to the
+cold tier. The cold tier is still a reponsive query tier but as the data transitions into this
+tier it can be compressed, shrunken, with zero replicas and backed by a <<ilm-searchable-snapshot, snapshot>>.
+These tier features offer a huge opprotunity to make major cost savings. Data withing this tier
+is retained for months and the indices have zero replicas as they are backed by a searchable
+snapshot.
+
+The tradeoff when moving data from Warm to Cold is writability for costs, with the search performance
+remaining the same as in the Warm tier.
+
+[[data-tier-allocation]]
+=== Data tier index allocation
+
+When an index is created {es} will automatically allocate the index to the <<content-tier, Content tier>>
+if the index is not part of a <<data-streams, data stream> or to the <<hot-tier, Hot tier>> if the index
+is part of a <<data-streams, data stream>>. The user created content will initially be allocated to the
+Content tier which is designed and optimised for this type of data. Timeseries data (ie. data streams)
+will initially be allocated to the Hot tier which offers high performance and is capable of efficiently
+ingesting large amounts of machine data (ie. logs, metrics) while also offering high query performance.
```

Comment:
I think we get into too much detail about performance characteristics here. Instead I'd rather have a blurb about exactly what setting is set by default (`index.routing.allocation.include._tier_preference`) and how to unset it or override it as mentioned below